src/common/ant.ts

```diff
@@ -76,6 +86,19 @@ export class AoANTReadable implements AoANTRead {
     const res = await this.process.read<AoANTState>({
       tags,
     });
+    const schemaResult = AntStateSchema.passthrough()
+      .and(
+        z.object({
+          Records: z.record(z.string(), AntRecordSchema.passthrough()),
+        }),
+      )
+      .safeParse(res);
+    if (!schemaResult.success) {
+      throw new Error(
+        'Invalid ANT State\n' +
+          JSON.stringify(schemaResult.error.format(), null, 2),
+      );
+    }
```

Comment:
nit, this could be a small utility function to format the error - either as a utility or private function on the class